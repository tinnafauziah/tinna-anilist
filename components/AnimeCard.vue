<template lang="pug">
v-tooltip(top)
  template(v-slot:activator='{ on, attrs }')
    v-card.mb-4.mr-4.anime_card(
      width='350',
      @click='redirectAnimeDetail()',
      v-bind='attrs',
      v-on='on')
      v-img(height='400', :src='anime.coverImage.large')
      v-card-text
        .subtitle-1.font-weight-bold.mb-1 {{ anime.title | animeTitle }}
        v-layout.mb-4(wrap, align-center)
          v-flex.xs1.mr-1
            v-icon(:color='getRatingColor(anime.averageScore)') {{ getRatingIcon(anime.averageScore) }}
          v-flex.xs10.mb-1
            span.subtitle-1 {{ anime.averageScore | animeAverageScore }}
        v-chip.mb-4.mr-4(v-for='genre in anime.genres', :key='genre') {{ genre }}
  span See Anime's Detail
</template>

<script>
import humanizeAnimeString from "~/mixins/humanize-anime-string";

export default {
  mixins: [humanizeAnimeString],
  name: "AnimeCard",
  props: {
    anime: {
      type: Object,
      default: () => ({
        id: null,
        coverImage: { large: null },
        title: {
          english: null,
          romaji: null,
        },
        averageScore: null,
        genres: [],
      }),
    },
  },
  methods: {
    redirectAnimeDetail() {
      this.$router.push(`/anime/${this.anime.id}`);
    },
  },
};
</script>
<style lang="scss" scoped>
.scroll_container {
  overflow: scroll;
  height: 586px;
}
</style>
