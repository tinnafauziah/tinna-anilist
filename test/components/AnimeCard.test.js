import AnimeCard from '~/components/AnimeCard';
import animeIndexData from '~/test/mocks/api-mocks/anime-list'

const mocks = {
  $router: {
    push: jest.fn(),
  },
};

const anime = animeIndexData.media[0];

describe('Anime Index Page', () => {
  const mockPage = (propsData = {}) => {
    return global.shallowMount(AnimeCard, { mocks, propsData });
  };
  it('should anime Index page match snapshot by default anime props', async () => {
    const wrapper = mockPage();
    expect(wrapper.element).toMatchSnapshot();
  });
  it('should anime Index page match snapshot', async () => {
    const wrapper = mockPage();
    await wrapper.setProps({ anime });
    expect(wrapper.element).toMatchSnapshot();
  });
  it('should redirect to detail', async () => {
    const wrapper = await mockPage({ anime });
    wrapper.vm.redirectAnimeDetail();

    expect(mocks.$router.push).toBeCalledWith(`/anime/${anime.id}`);
  });
});
