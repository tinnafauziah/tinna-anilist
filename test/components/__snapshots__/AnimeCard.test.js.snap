// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Anime Index Page should anime Index page match snapshot 1`] = `
<v-card-stub
  class="mb-4 mr-4 anime_card"
  loaderheight="4"
  tag="div"
  width="350"
>
  <v-img-stub
    height="400"
    options="[object Object]"
    position="center center"
    src="https://s4.anilist.co/file/anilistcdn/media/anime/cover/medium/bx127230-FlochcFsyoF4.png"
    transition="fade-transition"
  />
  <v-card-text-stub
    tag="div"
  >
    <div
      class="subtitle-1 font-weight-bold mb-1 teal--text darken-4"
    >
      Chainsaw Man
    </div>
    <v-layout-stub
      align-center="align-center"
      class="mb-4"
      tag="div"
      wrap="wrap"
    >
      <v-flex-stub
        class="xs1 mr-1"
        tag="div"
      >
        <v-icon-stub
          color="green"
        >
          mdi-emoticon-excited-outline
        </v-icon-stub>
      </v-flex-stub>
      <v-flex-stub
        class="xs10 mb-1"
        tag="div"
      >
        <span
          class="subtitle-1"
        >
          86%
        </span>
      </v-flex-stub>
    </v-layout-stub>
    <v-chip-stub
      active="true"
      activeclass=""
      class="mb-4 mr-4"
      closeicon="$delete"
      closelabel="$vuetify.close"
      filtericon="$complete"
      tag="span"
    >
      Action
    </v-chip-stub>
    <v-chip-stub
      active="true"
      activeclass=""
      class="mb-4 mr-4"
      closeicon="$delete"
      closelabel="$vuetify.close"
      filtericon="$complete"
      tag="span"
    >
      Comedy
    </v-chip-stub>
    <v-chip-stub
      active="true"
      activeclass=""
      class="mb-4 mr-4"
      closeicon="$delete"
      closelabel="$vuetify.close"
      filtericon="$complete"
      tag="span"
    >
      Drama
    </v-chip-stub>
    <v-chip-stub
      active="true"
      activeclass=""
      class="mb-4 mr-4"
      closeicon="$delete"
      closelabel="$vuetify.close"
      filtericon="$complete"
      tag="span"
    >
      Horror
    </v-chip-stub>
    <v-chip-stub
      active="true"
      activeclass=""
      class="mb-4 mr-4"
      closeicon="$delete"
      closelabel="$vuetify.close"
      filtericon="$complete"
      tag="span"
    >
      Supernatural
    </v-chip-stub>
  </v-card-text-stub>
</v-card-stub>
`;

exports[`Anime Index Page should anime Index page match snapshot by default anime props 1`] = `
<v-card-stub
  class="mb-4 mr-4 anime_card"
  loaderheight="4"
  tag="div"
  width="350"
>
  <v-img-stub
    height="400"
    options="[object Object]"
    position="center center"
    transition="fade-transition"
  />
  <v-card-text-stub
    tag="div"
  >
    <div
      class="subtitle-1 font-weight-bold mb-1 teal--text darken-4"
    >
      
    </div>
    <v-layout-stub
      align-center="align-center"
      class="mb-4"
      tag="div"
      wrap="wrap"
    >
      <v-flex-stub
        class="xs1 mr-1"
        tag="div"
      >
        <v-icon-stub
          color="gray"
        >
          mdi-emoticon-neutral-outline
        </v-icon-stub>
      </v-flex-stub>
      <v-flex-stub
        class="xs10 mb-1"
        tag="div"
      >
        <span
          class="subtitle-1"
        >
          no score yet
        </span>
      </v-flex-stub>
    </v-layout-stub>
  </v-card-text-stub>
</v-card-stub>
`;
